exp/controllers/awsmanagedmachinepool_controller.go

```diff
@@ -266,3 +325,187 @@ func managedControlPlaneToManagedMachinePoolMapFunc(c client.Client, gvk schema.
+		launchTemplateID, err := ec2svc.GetLaunchTemplateID(machinePoolScope.LaunchTemplateScope.Name())
+		if err != nil {
+			conditions.MarkUnknown(machinePoolScope.ManagedMachinePool, expinfrav1.LaunchTemplateReadyCondition, expinfrav1.LaunchTemplateNotFoundReason, err.Error())
+			return err
+		}
+		machinePoolScope.SetLaunchTemplateIDStatus(launchTemplateID)
+		return machinePoolScope.PatchObject()
+	}
+
+	if machinePoolScope.ManagedMachinePool.Status.LaunchTemplateVersion == nil {
+		launchTemplateVersion, err := ec2svc.GetLaunchTemplateLatestVersion(*machinePoolScope.ManagedMachinePool.Status.LaunchTemplateID)
+		if err != nil {
+			conditions.MarkUnknown(machinePoolScope.ManagedMachinePool, expinfrav1.LaunchTemplateReadyCondition, expinfrav1.LaunchTemplateNotFoundReason, err.Error())
+			return err
+		}
+		machinePoolScope.SetLaunchTemplateVersionStatus(launchTemplateVersion)
+		return machinePoolScope.PatchObject()
+	}
+
+	annotation, err := r.machinePoolAnnotationJSON(machinePoolScope.ManagedMachinePool, TagsLastAppliedAnnotation)
+	if err != nil {
+		return err
+	}
+
+	// Check if the instance tags were changed. If they were, create a new LaunchTemplate.
+	tagsChanged, _, _, _ := tagsChanged(annotation, machinePoolScope.LaunchTemplateScope.AdditionalTags()) // nolint:dogsled
+
+	needsUpdate, err := ec2svc.LaunchTemplateNeedsUpdate(machinePoolScope.LaunchTemplateScope, machinePoolScope.LaunchTemplateScope.AWSLaunchTemplate, launchTemplate)
+	if err != nil {
+		return err
+	}
+
+	// Create a new launch template version if there's a difference in configuration, tags,
+	// userdata, OR we've discovered a new AMI ID.
+	if needsUpdate || tagsChanged || *imageID != *launchTemplate.AMI.ID || launchTemplateUserDataHash != bootstrapDataHash {
+		machinePoolScope.Info("creating new version for launch template", "existing", launchTemplate, "incoming", machinePoolScope.LaunchTemplateScope.AWSLaunchTemplate)
+		// There is a limit to the number of Launch Template Versions.
+		// We ensure that the number of versions does not grow without bound by following a simple rule: Before we create a new version, we delete one old version, if there is at least one old version that is not in use.
+		if err := ec2svc.PruneLaunchTemplateVersions(*machinePoolScope.ManagedMachinePool.Status.LaunchTemplateID); err != nil {
+			return err
+		}
+		if err := ec2svc.CreateLaunchTemplateVersion(machinePoolScope.ManagedMachinePool.Status.LaunchTemplateID, machinePoolScope.LaunchTemplateScope, imageID, bootstrapData); err != nil {
+			return err
+		}
+		version, err := ec2svc.GetLaunchTemplateLatestVersion(*machinePoolScope.ManagedMachinePool.Status.LaunchTemplateID)
+		if err != nil {
+			return err
+		}
+		machinePoolScope.SetLaunchTemplateVersionStatus(version)
+		return machinePoolScope.PatchObject()
+	}
+
+	return nil
+}
+
+func (r *AWSManagedMachinePoolReconciler) reconcileTags(machinePoolScope *scope.ManagedMachinePoolScope, ec2Scope scope.EC2Scope) error {
```

Comment:
Same comment about reusing the logic between the 2 different machine pool controllers

Reply:
Moved the logic into the launchtemplate service as well.